File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/_tokens/spacing/SpacingRoles.kt
@file:Suppress("ktlint:standard:package-name")

package com.perrystreet.woof.designsystem.atomic._tokens.spacing

import androidx.compose.ui.unit.Dp
import com.perrystreet.woof.designsystem.atomic._primitives.spacing.SpacingPrimitives

object SpacingRoles {
    enum class Component(val dp: Dp) {
        Hairline(SpacingPrimitives.Space2),
        ExtraCompact(SpacingPrimitives.Space4),
        Compact(SpacingPrimitives.Space8),
        Cozy(SpacingPrimitives.Space12),
        Regular(SpacingPrimitives.Space16),
        Relaxed(SpacingPrimitives.Space20),
        Expanded(SpacingPrimitives.Space24),
        ExtraExpanded(SpacingPrimitives.Space32),
    }

    enum class Module(val dp: Dp) {
        Compact(SpacingPrimitives.Space20),
        Regular(SpacingPrimitives.Space40),
        Expanded(SpacingPrimitives.Space60),
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/_Tokens/Spacing/SpacingRoles.swift
import Foundation

public enum SpacingRoles {
    public enum Component {
        case hairline
        case extraCompact
        case compact
        case cozy
        case regular
        case relaxed
        case expanded
        case extraExpanded

        public var rawValue: CGFloat {
            switch self {
            case .hairline: SpacingPrimitives.space2.rawValue
            case .extraCompact: SpacingPrimitives.space4.rawValue
            case .compact: SpacingPrimitives.space8.rawValue
            case .cozy: SpacingPrimitives.space12.rawValue
            case .regular: SpacingPrimitives.space16.rawValue
            case .relaxed: SpacingPrimitives.space20.rawValue
            case .expanded: SpacingPrimitives.space24.rawValue
            case .extraExpanded: SpacingPrimitives.space32.rawValue
            }
        }
    }

    public enum Module {
        case compact
        case regular
        case expanded

        public var rawValue: CGFloat {
            switch self {
            case .compact: SpacingPrimitives.space20.rawValue
            case .regular: SpacingPrimitives.space40.rawValue
            case .expanded: SpacingPrimitives.space60.rawValue
            }
        }
    }
}
